File: problems/237/Lc237.ts
/**
 * Definition for singly-linked list.
 */
class ListNode {
  val: number
  next: ListNode | null
  constructor(val?: number, next?: ListNode | null) {
    this.val = (val === undefined ? 0 : val)
    this.next = (next === undefined ? null : next)
  }
}

/**
 Do not return anything, modify it in-place instead.
 */
function deleteNode(node: ListNode | null): void {
  // 1.
  // if (node !== null) {
  //   const nextNode = node.next;
  //   if (nextNode !== null) {
  //     node.val = nextNode.val;
  //     node.next = nextNode.next;
  //   }
  // }

  // 2.
  // if (node !== null && node.next !== null) {
  //   const nextNode = node.next;
  //   node.val = nextNode.val
  //   node.next = nextNode.next;
  // }

  // 3.
  if (node !== null && node.next !== null) {
    node.val = node.next.val
    node.next = node.next.next;
  }
};

File: problems/237/Lc237.java
public class Lc237 {
  static Lc237 leetcode = new Lc237();
  static Solution sol = leetcode.new Solution();

  /**
   * Definition for singly-linked list.
   */
  public class ListNode {
    int val;
    ListNode next;

    ListNode(int x) {
      val = x;
    }
  }

  class Solution {
    public void deleteNode(ListNode node) {
      node.val = node.next.val;
      node.next = node.next.next;

      // ListNode nextNode = node.next;
      // node.val = nextNode.val;
      // node.next = nextNode.next;
    }
  }

  static void testcase01() {
    // assert expected.equals(result) : "expected" + String.valueOf(expected) + ",
    // but " + String.valueOf(result);
  }

  public static void main(String[] args) {
    testcase01();
  }
}
